### DISCUSSION TITLE
Server api detect ssr

### DISCUSSION THREAD

**Discussion by ColonelBundy** [nuxt/nuxt#16564] [Questions]
When defining an api route using `defineEventHandler` is it possible to determine if the route is invoked during ssr or from the client?

`process.server` will always be true so that won't work.

**Comment by TheAlexLichter** [comment]
The best way would be probably some kind of `secret` defined via `runtimeConfig` (private). This shouldn't appear in the source code and both, the endpoint and the application during SSR know about it. You can pass it e.g. via a custom header then ☺️
